cartographer_ros/cartographer_ros/node.cc

```diff
@@ -220,6 +228,13 @@ void Node::PublishTrajectoryStates(const ::ros::WallTimerEvent& timer_event) {
   }
 }
 
+void Node::PublishTrajectoryNodesList(
+    const ::ros::WallTimerEvent& unused_timer_event) {
+  carto::common::MutexLocker lock(&mutex_);
+  trajectory_nodes_list_publisher_.publish(
```

Comment:
We should first measure if this is important. If it is, we already do this before calling `BuildOccupancyGrid` and could handle this here similarly.